lib/ddtrace/contrib/grape/endpoint.rb

```diff
@@ -196,6 +204,58 @@ def analytics_sample_rate
             datadog_configuration[:analytics_sample_rate]
           end
 
+          def error_responses
+            return datadog_configuration[:error_responses] if datadog_configuration[:error_responses].is_a?(String) && !status.nil?
+            datadog_configuration[:error_responses].join(',') if status.is_a?(Array) && !status.empty?
+          end
+
+          def handle_statuses
+            if error_responses
+              error_responses.gsub(/\s+/, '').split(',').select do |code|
+                if !code.to_s.match(/^\d{3}(?:-\d{3})?(?:,\d{3}(?:-\d{3})?)*$/)
+                  Datadog.logger.debug("Invalid config provided: #{code}. Must be formatted like '400-403,405,410-499'.")
+                  next
+                else
+                  true
+                end
+              end
+            else
+              Datadog.logger.debug('No valid config was provided for :error_responses - falling back to default.')
+              ['500-599'] # Rather than returning an empty array, we need to fallback to default config.
+            end
+          end
+
+          def set_range
+            @datadog_set ||= begin
+              set = Set.new
+              handle_statuses.each do |statuses|
+                status = statuses.to_s.split('-')
+                if status.length == 1
+                  set.add(Integer(status[0]))
+                elsif status.length == 2
+                  min, max = status.minmax
+                  Array(min..max).each do |i|
+                    set.add(Integer(i))
+                  end
+                end
+              end
+              set
+            end
+
+            @datadog_set
+          end
+
+          def exception_is_error?(exception)
+            status = nil
+            return false unless exception
+            if exception.respond_to?('status') && set_range.include?(exception.status)
```

Comment:
After https://github.com/DataDog/dd-trace-rb/pull/1238/files#r520721404 is implemented, `datadog_configuration[:error_responses]` will return the matcher object `StatusCodeMatcher`, and we'll be able to directly call it:

```ruby
matcher = datadog_configuration[:error_responses]
if exception.respond_to?('status') && matcher && matcher.include?(exception.status)
```